src/Swift.Bindings/src/Demangler/Swift5Demangler.cs

```diff
@@ -0,0 +1,2961 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+#nullable disable
+
+// This is a port of the Apple Swift 5 demangler
```

Comment:
Is this ported from the old BTfS? 

You mentioned in the PR description that the cost of changing this will be high. Based on the name I suspect the demangler is Swift5 specific. With the Swift6 release, do you know if there are any necessary changes that we will need to do?

Reply:
This is moved from BTfS, yes. I believe I read a few years back that Apple has committed to this format moving forward. That doesn't mean that there won't be changes, but they need to be backwards compatible to this because of library evolution. And, as it turns out, since part of the reason why this is being used is to be able to get the thunks for instance methods, if they changed the way those work, they would be breaking everything pre 6 with library evolution.